DESIGN.md

```diff
@@ -325,18 +325,158 @@ In the addition of the previous rules, an operator for `Flowable`:
 
 ### Creation
 
-Creation of the various types should be exposed through factory methods that provide safe construction.
+Unlike RxJava 1.x, 2.x base classes are to be abstract, stateless and generally no longer wrap an `OnSubscribe` callback - this saves allocation in assembly time without limiting the expressiveness. Operator methods and standard factories still live as final on the base classes.
+
+Instead of the indirection of an `OnSubscribe` and `lift`, operators are to be implemented by extending the base classes. For example, the `map`
```

Comment:
> but you won't be able to combine FlowableMap and ObservableMap because the actual output class differs and needs different calls to onSubscribe.

@akarnokd I am telling you that is not what I am saying! Read my comments carefully! I am saying that a Map operator can be written that works over all Flowables AND a separate map operator can be written that works over all `Observables`! I really don't know why you don't read my comments. 

> but I don't see why unless you want to encourage other libraries to implement them differently than in RxJava.

We talked about this when you tried to implement `Consumable`. I thought you were on board with creating a base interface or class for all kinds of Observables (and likewise a different base for Completables and Singles). Is there something I'm missing?

> The plan

Who's plan? I know that I'm pretty confused about the perceived benefits here. It seems like a micro optimization to save an allocation.